brightener.h

```diff
@@ -4,5 +4,6 @@
 using namespace std;
 
 shared_ptr<Image> BrightenWholeImage(shared_ptr<Image> inputImage, int& attenuatedPixelCount);
```

Comment:
`attenuatedPixelCount` is a non-const reference. Consider moving it.
Perhaps it helps if we move it into a place where it belongs... it's the number of pixels attenuated when brightening the image. What if we keep it together with `BrightenedImage`?